src/sortStudents.ts

```diff
@@ -1,16 +1,68 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'averageGrade',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function calcAvg(student: Student): number {
+  return student.grades
+    .reduce((acum: number, grade: number) => acum + grade, 0)
+   / student.grades.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const copiedStudents = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      return copiedStudents
+        .sort((aStudent, bStudent) => {
+          if (order === 'asc') {
+            return aStudent[sortBy].localeCompare(bStudent[sortBy]);
+          }
+
+          return bStudent[sortBy].localeCompare(aStudent[sortBy]);
+        });
+
+    case SortType.Age:
+    case SortType.Married:
+      if (order === 'asc') {
+        return copiedStudents
+        // eslint-disable-next-line
+        .sort((aStudent, bStudent) => Number(aStudent[sortBy]) - Number(bStudent[sortBy]));
+      }
+
+      return copiedStudents
+      // eslint-disable-next-line
+        .sort((aStudent, bStudent) => Number(bStudent[sortBy]) - Number(aStudent[sortBy]));
```

Comment:
you can make like that
```suggestion
        .sort((aStudent, bStudent) => order === 'asc' 
          ? Number(aStudent[sortBy]) - Number(bStudent[sortBy]));
          : Number(bStudent[sortBy]) - Number(aStudent[sortBy]))
```